includes/Checker/AJAX_Runner.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Class WordPress\Plugin_Check\Checker\AJAX_Runner
+ *
+ * @package plugin-check
+ */
+
+namespace WordPress\Plugin_Check\Checker;
+
+use WordPress\Plugin_Check\Utilities\Plugin_Request_Utility;
+
+/**
+ * AJAX Runner class.
+ *
+ * @since n.e.x.t
+ */
+class AJAX_Runner extends Abstract_Check_Runner {
+
+	/**
+	 * An instances of the Checks class.
+	 *
+	 * @since n.e.x.t
+	 * @var Checks
+	 */
+	protected $checks;
+
+	/**
+	 * Checks if the current request is an AJAX request for the Plugin Checker.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return bool
+	 */
+	public function is_plugin_check() {
+		if ( 0 !== strpos( $_SERVER['REQUEST_URI'], '/admin-ajax.php' ) ) {
```

Comment:
WordPress has a function for that which is more reliable and we should probably use instead: `wp_doing_ajax()`